indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalReplaceAction.java

```diff
@@ -140,26 +153,29 @@ public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox)
               )
               .build()
       );
+
+      IndexTaskUtils.emitSegmentPublishMetrics(publishResult, task, toolbox);
+
+      // Upgrade any overlapping pending segments
+      // Do not perform upgrade in the same transaction as replace commit so that
+      // failure to upgrade pending segments does not affect success of the commit
+      if (publishResult.isSuccess() && toolbox.getSupervisorManager() != null) {
+        try {
+          registerUpgradedPendingSegmentsOnSupervisor(task, toolbox, publishResult.getUpgradedPendingSegments());
+        }
+        catch (Exception e) {
+          log.error(e, "Error while upgrading pending segments for task[%s]", task.getId());
+        }
```

Comment:
It feels a little weird to have this try-catch inside the outer try-catch.
Maybe we could move this try-catch into the method `registerUpgradePendingSegments...` and rename it to `tryRegisterUpgradePendingSegments...` to denote that it throws no exceptions.

Reply:
Done